tests/ha/single_pod_restart/test_server_pod_restart_api.py

```diff
@@ -1279,6 +1279,135 @@ def test_obj_ver_after_server_pod_restart(self):
+                                               self.object_name, **args)
+        assert_utils.assert_true(resp[0], f"Upload Object failed {resp[1]}")
+        self.version_etag[self.bucket_name].extend(resp[1])
+        resp = self.ha_obj.list_verify_version(self.s3_ver, self.bucket_name, self.version_etag[
+            self.bucket_name])
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 2: Uploaded same object %s after enabling versioning and listed & "
+                    "verified the same.", self.object_name)
+
+        LOGGER.info("Step 3: Shutdown server pod with replica method and verify cluster & "
+                    "remaining pods status")
+        resp = self.ha_obj.delete_kpod_with_shutdown_methods(
+            master_node_obj=self.node_master_list[0], health_obj=self.hlth_master_list[0],
+            pod_prefix=[const.SERVER_POD_NAME_PREFIX], delete_pod=[self.delete_pod],
+            num_replica=self.num_replica - 1)
+        # Assert if empty dictionary
+        assert_utils.assert_true(resp[1], "Failed to shutdown/delete server pod")
+        pod_name = list(resp[1].keys())[0]
+        self.set_name = resp[1][pod_name]['deployment_name']
+        self.restore_pod = True
+        self.restore_method = resp[1][pod_name]['method']
+        assert_utils.assert_true(resp[0], "Cluster/Services status is not as expected")
+        LOGGER.info("Step 3: Successfully shutdown server pod %s. Verified cluster and services "
+                    "states are as expected & remaining pods status is online.", pod_name)
+
+        LOGGER.info("Step 4: Get object with version IDs and verify etags.")
+        resp = self.ha_obj.parallel_get_object(event=event, s3_ver_obj=self.s3_ver,
+                                               bkt_name=self.bucket_name, obj_name=self.object_name,
+                                               ver_etag=self.version_etag[self.bucket_name])
+        assert_utils.assert_true(resp[0], f"Get Object with versionID failed {resp[1]}")
+        LOGGER.info("Step 4: Got object with version IDs and verify etags.")
+
+        LOGGER.info("Step 5: Upload same object %s after server pod shutdown and list & verify "
+                    "the same.", self.object_name)
+        args = {'file_path': self.multipart_obj_path}
+        resp = self.ha_obj.parallel_put_object(event, self.s3_test_obj, self.bucket_name,
+                                               self.object_name, **args)
+        assert_utils.assert_true(resp[0], f"Upload Object failed {resp[1]}")
+        self.version_etag[self.bucket_name].extend(resp[1])
+        resp = self.ha_obj.list_verify_version(self.s3_ver, self.bucket_name, self.version_etag[
+            self.bucket_name])
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 5: Uploaded same object %s after server pod shutdown and listed & "
+                    "verified the same.", self.object_name)
```

Comment:
You are listing and verifying version IDs, can you add that in logger.
Check at all the places.